core/service.go

```diff
@@ -323,21 +331,25 @@ func (s *Service) handleReceivedMessage(msg network.Message) (err error) {
 // announce messages (block announce messages include the header but the full
 // block is required to execute `core_execute_block`).
 func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
+	data := msg.(*network.BlockAnnounceMessage)
 
-	// TODO: check if we should send block request message
+	header, err := types.NewHeader(data.ParentHash, data.Number, data.StateRoot, data.ExtrinsicsRoot, data.Digest)
+	if err != nil {
+		return err
+	}
 
-	// TODO: update message properties and use generated id
-	blockRequest := &network.BlockRequestMessage{
-		ID:            1,
-		RequestedData: 2,
-		StartingBlock: []byte{},
-		EndBlockHash:  optional.NewHash(true, common.Hash{}),
-		Direction:     1,
-		Max:           optional.NewUint32(false, 0),
+	// TODO: check if we should send block request message
```

Comment:
Can we resolve this here? We just need to see if the announce message is later than the latest block in state then send the request message to the network service.

Reply:
isn't that what Thomas is working on? I could do it but I think it's out of scope

Reply:
right... I was confused by the issue and the title of the pull request, which is related to sending a block request based on the status message, not the block announce message.

The two pull requests overlap quite a bit. Might be best to wait with the pull request.

Reply:
okay it would be good to get this one in because this allows for non authority nodes to sync a block producer